File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int ans=0,minL=prices[0];
        for(int i=0;i<n;i++){
            ans=max(ans,prices[i]-minL);
            minL = min(minL,prices[i]);
        }
        return ans;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let minTill = prices[0];
    let ans = 0;
    for (let i = 1; i < prices.length; i++) {
        ans = Math.max(ans, prices[i] - minTill);
        minTill = Math.min(minTill, prices[i])
    }
    return ans;
};